src/metrics/misc.ts

```
@@ -17,3 +26,163 @@ export function countGamesInDataset(datasetPath: string): number {
+    this.mostGamesPlayed = maxGames;
+    this.playerMostGames = playerMostGames;
+
+    return {
+      averagePlayerRating: this.averagePlayerRating,
+      averageRatingDiff: this.averageRatingDiff,
+      mostGamesPlayed: maxGames,
+      playerMostGames,
+    };
+  }
+
+  // Reset the maps used to track metrics
+  clear(): void {
+    this.numberGamesAnalyzed = 0;
+    this.averagePlayerRating = 0;
+    this.averageRatingDiff = 0;
+    this.largestRatingDiff = 0;
+    this.largestRatingDiffGame = '';
+    this.mostGamesPlayed = 0;
+    this.playerMostGames = '';
+    this.gameTypeStats = {
+      numberBulletGames: 0,
+      numberBlitzGames: 0,
+      numberRapidGames: 0,
+      numberClassicalGames: 0,
+    };
+    this.gameTimeControlStats = {};
+
+    this.totalPlayerRating = 0;
+    this.totalPlayerRatingDiff = 0;
+    this.playerGameStats = {};
+
+  }
+
+  processGame(
+    game: { move: PrettyMove; board: Piece[] }[],
+    metadata?: string[]
+  ) {
+
+    // Update the gameTimeControlStats based on the time control of the game
+    const timeControl = metadata?.find((data) => data.startsWith('[TimeControl'));
+    if (timeControl) {
+      const timeControlQuantity = timeControl.split(' ')[1];
+      if (this.gameTimeControlStats[timeControlQuantity]) {
+        this.gameTimeControlStats[timeControlQuantity]++;
+      } else {
+        this.gameTimeControlStats[timeControlQuantity] = 1;
+      }
+    }
+    
+    // Identify the time control type from the metadata
+    const gameType = metadata?.find((data) => data.startsWith('[Event'));
+    if (gameType) {
+      if (gameType.includes('Bullet')) {
+        this.gameTypeStats.numberBulletGames++;
+      } else if (gameType.includes('Blitz')) {
+        this.gameTypeStats.numberBlitzGames++;
+      } else if (gameType.includes('Rapid')) {
+        this.gameTypeStats.numberRapidGames++;
+      } else if (gameType.includes('Classical')) {
+        this.gameTypeStats.numberClassicalGames++;
+      }
```

can we use an exact match instead of the string search method?

---

alternative is something like this but im worried with split that if we get an unexpected string in event (like "unrated tournament blitz game" or something like that it wouldn't work. so i'd rather search (can test speed later if need be)

const eventString = metadata?.find((data) => data.startsWith('[Event'));
if (eventString) {
  const gameType = eventString.split('"')[1].split(' ')[1];
  if (gameType === 'Bullet') {
    this.gameTypeStats.numberBulletGames++;

---

wouldn't we want to throw an error if we get a new game type?

---

good pt, will add in